src/TsGenerator.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+declare(strict_types=1);
+
+namespace app;
+
+class TsGenerator
+{
+    public const OP_API_OPERATION_MAP = 'api_operation_map';
+    public const OP_MODEL = 'model';
+    public const OP_GENERAL = 'general';
+
+    public function __construct(
+        private Config $config,
+    ) {}
+
+    public static function allowedTypes(): array
+    {
+        return [
+            static::OP_API_OPERATION_MAP,
+            static::OP_MODEL,
+        ];
+    }
+
+    public function generate(array $data, string $type, string $output): void
+    {
+        match ($type) {
+            static::OP_API_OPERATION_MAP => $this->generateApiOperationMap($data, $output),
+            static::OP_MODEL => $this->generateModel($data, $output),
+            static::OP_GENERAL => $this->generateGeneral($data, $output),
+        };
+    }
+
+    protected function generateApiOperationMap(array $data, string $output): void
+    {
+        printInfo('Generating ApiOperationMap...');
+
+        $paths = $data['paths'] ?? [];
+        $operationItems = '';
+        $operationItemsTemplate = file_get_contents('stubs/api_operation_map/api_operation_map_item.stub');
+
+        foreach ($paths as $path => $pathInfo) {
+            if (! is_array($pathInfo)) {
+                continue;
+            }
+
+            foreach ($pathInfo as $method => $info) {
+                $operationId = $info['operationId'];
+
+                if (! $operationId) {
+                    continue;
+                }
+
+                $operationItems .= str_replace(
+                    ['{operationId}', '{url}', '{method}'],
+                    [$operationId, $path, strtoupper($method)],
+                    $operationItemsTemplate,
```

Comment:
new request from FE:
they need to identify the request is suppose to return single or multiple entries, so inside object need to add one more key:
```
paginateBankAccounts: { url: '/v2/bank-accounts', method: 'POST', responseType: 'collection'},
```

responseType: resource|collection

this may be defined as:
- in case we have ref (in may be complex structure with $ref=any of (successResponse + data: $ref=Resource|Collection) so should support this - we check suffix Resource or Collection
- in case no suffix - we need to check is data: is type object or type: array